File: src/MediCore.Desktop/lib/features/encounters/domain/entities/encounter.dart
class Encounter {
  final String id;
  final String patientId;
  final String doctorId;
  final String? appointmentId;
  final String encounterType;
  final String status;
  final String? chiefComplaint;
  final String? hpi;
  final String? physicalExam;
  final String? clinicalNotes;
  final String? treatmentPlan;
  final DateTime? followUpDate;
  final String? followUpNotes;
  final bool isLocked;
  final DateTime createdAt;

  const Encounter({
    required this.id,
    required this.patientId,
    required this.doctorId,
    this.appointmentId,
    required this.encounterType,
    required this.status,
    this.chiefComplaint,
    this.hpi,
    this.physicalExam,
    this.clinicalNotes,
    this.treatmentPlan,
    this.followUpDate,
    this.followUpNotes,
    required this.isLocked,
    required this.createdAt,
  });
}

File: src/MediCore.API/MediCore.Domain/Entities/Medical/Encounter.cs
using MediCore.Domain.Common;

namespace MediCore.Domain.Entities.Medical;

public class Encounter : BaseEntity
{
    public Guid TenantId { get; set; }
    public Guid BranchId { get; set; }
    public Guid PatientId { get; set; }
    public Guid DoctorId { get; set; }
    public Guid? AppointmentId { get; set; }
    public string EncounterType { get; set; } = "outpatient";
    public string Status { get; set; } = "in_progress";
    public string? ChiefComplaint { get; set; }
    public string? Hpi { get; set; }
    public string? PhysicalExam { get; set; }
    public string? ClinicalNotes { get; set; }
    public string? TreatmentPlan { get; set; }
    public DateOnly? FollowUpDate { get; set; }
    public string? FollowUpNotes { get; set; }
    public bool IsLocked { get; set; } = false;
    public DateTime? LockedAt { get; set; }

    // Navigation Properties
    public Patient Patient { get; set; } = null!;
    public Branch Branch { get; set; } = null!;
    public Appointment? Appointment { get; set; }
}